webapp/graphql/PostQuery.js

```diff
@@ -128,3 +42,22 @@ export const PostsEmotionsByCurrentUser = () => {
     }
   `
 }
+
+export const relatedContributions = i18n => {
+  const lang = i18n.locale().toUpperCase()
+  return gql`
+    ${postFragment(lang)}
+    ${postCountsFragment}
+
+    query Post($slug: String!) {
+      Post(slug: $slug) {
+        ...post
+        ...postCounts
+        relatedContributions(first: 2) {
+          ...post
+          ...postCounts
+        }
+      }
+    }
+  `
+}
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-03T12:06:50Z" title="Tuesday, September 3rd 2019, 2:06:50 pm +02:00">Sep 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -128,3 +128,51 @@ export const PostsEmotionsByCurrentUser = () => {
     }
   `
 }
+
+export const relatedContributions = i18n => {
+  const lang = i18n.locale().toUpperCase()
+  return gql`query Post($slug: String!) {
+    Post(slug: $slug) {
+      id
+      title
+      tags {
+        id
+      }
+      categories {
+        id
+        name
+        icon
+      }
+      relatedContributions(first: 2) {
+        id
+        title
+        slug
+        contentExcerpt
+        shoutedCount
+        categories {
+          id
+          name
+          icon
+        }
+        author {
+          id
+          name
+          slug
+          avatar
+          contributionsCount
+          followedByCount
+          followedByCurrentUser
+          commentedCount
+          location {
+            name: name${lang}
+          }
+          badges {
+            id
+            icon
+          }
+        }
+      }
+      shoutedCount
+    }
+  }`
+}
```


never seen this before, not clear what it does... returns two relatedContributions, do you have a link to some docs on this?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-03T12:07:35Z" title="Tuesday, September 3rd 2019, 2:07:35 pm +02:00">Sep 3, 2019</time>_

---


oh, sorry I can see it was already in the code base...

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-03T13:01:18Z" title="Tuesday, September 3rd 2019, 3:01:18 pm +02:00">Sep 3, 2019</time>_

---


Yes, it's one of the bugfixes along the way. There was `commented` requested on type `Post` which does not make a lot of sense. To reduce the size of the queries, we could potentially use fragments.